subsystems/drivebase/SwerveDrive.java

```diff
@@ -311,8 +329,43 @@ public void mustangPeriodic() {
         Logger.recordOutput(DRIVEBASE_PITCH, getPitch());
         Logger.recordOutput(DRIVEBASE_ROLL, getRoll());
 
+        if (kConfig.kDriveMotorType.equals(Motor_Type.KRAKEN_X60)) {
+            double latestDriveCurrentDraw = 0;
+            for (int i = 0; i < mModules.length; i++) {
+                latestDriveCurrentDraw += ((TalonFX) mModules[i].getDriveMotor()).getSupplyCurrent();
+            }
+
+            double currentTime = System.currentTimeMillis() / 1000;  // Current time in seconds
+            driveCurrentBuffer.add(new CurrentMeasurement(currentTime, latestDriveCurrentDraw));
+            
+            // Remove old measurements from the buffer.
+            while (!driveCurrentBuffer.isEmpty() && currentTime - driveCurrentBuffer.peek().getTime() > 5) {
+                driveCurrentBuffer.remove();
+            }
+            
+            // Calculate the average current draw over the rolling window
+            double totalCurrent = 0;
+            int count = 0;
+            for (CurrentMeasurement measurement : driveCurrentBuffer) {
+                totalCurrent += measurement.getCurrentDraw();
+                count++;
+            }
+            double averageCurrent = totalCurrent / count;
+            
+            // Check if average current exceeds the threshold
+            if ((averageCurrent > 120 && currentLimitConfigs.SupplyCurrentLimit != 25) || (averageCurrent <= 120 && currentLimitConfigs.SupplyCurrentLimit != this.kConfig.kMaxDriveCurrent)) {
```

Comment:
Let's move the magic numbers into the constants file so we always go to the one central place to tweak this stuff as needed